spec/support/features/interaction_helper.rb

```diff
@@ -1,19 +1,22 @@
 module InteractionHelper
   def click_spinner_button_and_wait(...)
     click_on(...)
-    begin
+    wait_for_content_to_disappear do
       expect(page).to have_no_css('lg-spinner-button.spinner-button--spinner-active', wait: 10)
-    rescue Selenium::WebDriver::Error::StaleElementReferenceError,
-           Selenium::WebDriver::Error::UnknownError => e
-      raise e if e.is_a?(Selenium::WebDriver::Error::UnknownError) &&
-                 !e.message.include?('Node with given id does not belong to the document')
-      # A stale element error can occur when attempting to wait for the spinner to disappear if the
-      # context in which the button was clicked (e.g. a `within` block) itself disappears. This is
-      # fine, since if the ancestor disappears, it can be assumed that the button is gone too.
-      #
+    end
+  end
+
+  def wait_for_content_to_disappear do
+    begin
+      yield
+    rescue Selenium::WebDriver::Error::StaleElementReferenceError
+      # A StaleElementReferenceError means that the context the element
+      # was in has disappeared, which means the element is gone too.
+    rescue Selenium::WebDriver::Error::UnknownError => e
       # We sometimes see "UnknownError" with an error message that is similar to a
       # StaleElementReferenceError, but have not been able to resolve it and are ignoring it
       # for now.
+      raise e if !e.message.include?('Node with given id does not belong to the document')
     end
   end
```

Comment:
big fan! we can use the top-level rescue to remove some indentation

```suggestion
  def wait_for_content_to_disappear do
    yield
  rescue Selenium::WebDriver::Error::StaleElementReferenceError
    # A StaleElementReferenceError means that the context the element
    # was in has disappeared, which means the element is gone too.
  rescue Selenium::WebDriver::Error::UnknownError => e
    # We sometimes see "UnknownError" with an error message that is similar to a
    # StaleElementReferenceError, but have not been able to resolve it and are ignoring it
    # for now.
    raise e if !e.message.include?('Node with given id does not belong to the document')
  end
```